fearless/Common/Services/ExtrinsicService/ExtrinsicService.swift

```diff
@@ -239,6 +375,35 @@ extension ExtrinsicService: ExtrinsicServiceProtocol {
         operationManager.enqueue(operations: wrapper.allOperations, in: .transient)
     }
 
+    func estimateFee(
+        _ closure: @escaping ExtrinsicBuilderIndexedClosure,
+        runningIn queue: DispatchQueue,
+        numberOfExtrinsics: Int,
+        completion completionClosure: @escaping EstimateFeeIndexedClosure
+    ) {
+        let wrapper = operationFactory.estimateFeeOperation(
+            closure,
+            numberOfExtrinsics: numberOfExtrinsics
+        )
+
+        wrapper.targetOperation.completionBlock = {
+            queue.async {
+                if let operationResult = wrapper.targetOperation.result {
```

Comment:
Please, use extractNonCancellable result to avoid duplication code for error handling